### PR TITLE
Remove atmos thermo wrappers

### LINKED ISSUE #1544: Remove moisture thermo wrappers
# Description

We should remove the `moisture.jl` wrappers:

 - `temperature`
 - `pressure`
 - `soundspeed`

because they assume a particular way that the thermo state is constructed (and it's safer/more efficient to call the method directly with a given thermo state). Also, this is not really scalable since we'd generally need to add wrappers for every thermo function. See discussion [here](https://github.com/CliMA/ClimateMachine.jl/pull/1528#discussion_r485281216).

## Additional context

Add any other reasons why this should be addressed.

# For CLIMA Developers

- [ ] This feature can be added (if it cannot be, explain why in a comment below -- lack of technical expertise, not relevant to the scope of this project, too ambitious)
- [ ] There is a timeline for when this feature can be implemented
- [ ] The feature has been (or will be) implemented (Please link the PR)

### PR DESCRIPTION
**Pull request by charleskawczynski**
# Description

This PR
 - Changes `src/Atmos/Model/thermodynamics.jl` to `src/Atmos/Model/thermo_states.jl` to improve stack traces (avoiding confusion with `src/Common/Thermodynamics/Thermodynamics.jl`)
 - Removes
    - `pressure`
    - `temperature`
    - `total_specific_enthalpy`

from `src/Atmos/Model/moisture.jl`. Closes #1544, #1381.

I have

- [ ] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [ ] Followed all necessary [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and run `julia .dev/climaformat.jl .`
- [ ] Updated the documentation to reflect changes from this PR.

# For review by CLIMA developers

- [ ] There are no open pull requests for this already
- [ ] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

File: src/Atmos/Model/AtmosModel.jl [hunk 1]
```diff
@@ -27,7 +27,7 @@ using ..TurbulenceClosures
 import ..TurbulenceClosures: turbulence_tensors
 using ..TurbulenceConvection
 
-import ..Thermodynamics: internal_energy, total_specific_enthalpy
+import ..Thermodynamics: internal_energy
```
Comment (trontrytel) [inline src/Atmos/Model/AtmosModel.jl:30]: Doesn't look like `internal_energy` is being used either?

Comment (trontrytel) [reply src/Atmos/Model/AtmosModel.jl:30]: No, sorry. It is used later in `moisture.jl`

**Review by trontrytel — APPROVED** [review]
LGTM!

**Review by mwarusz — APPROVED** [review]

**Comment by trontrytel** [comment]
fancy meeting you here @mwarusz  ;)

**Pull request merged**